assembl/models/discussion.py

```diff
@@ -31,12 +31,51 @@
 from .langstrings import LangString
 from .preferences import Preferences
 from assembl.lib.caching import create_analytics_region
-
+from assembl.graphql import docstrings as docs
 resolver = DottedNameResolver(__package__)
 log = logging.getLogger()
 visit_analytics_region = create_analytics_region()
 
 
+class OldSlug(DiscussionBoundBase, NamedClassMixin):
+    """
+    The old slug of a debate. Used in case the administrator changes the slug of a debate.
+    """
+    __tablename__ = "old_slug"
+    id = Column(Integer, primary_key=True)
+    discussion_id = Column(Integer, ForeignKey(
+        'discussion.id',
+        ondelete='CASCADE',
+        onupdate='CASCADE'),
+        nullable=False,
+        index=True)
+    discussion = relationship(
+        "Discussion",
+        backref=backref(
+            'old_slugs',
+            cascade="all, delete-orphan"),
+    )
+    slug = Column(CoerceUnicode, nullable=False, doc=docs.OldSlug.slug)
+    redirection_slug = Column(CoerceUnicode, doc=docs.OldSlug.redirection_slug)
```

Comment:
I don't understand your chosen structure.
I would have kept only the old slug in this table, not the old and new.
If you rename
a -> b
b -> c
you should have in discussion.old_slugs=[OldSlug('a'), OldSlug('b')]

You should be able to just look up 'a' in OldSlug and get oldslug.discussion.slug to retrieve 'c'
Your `find_discussion_from_slug` should be much simpler.